animation_nodes/nodes/rotation/convert_rotations.py

```diff
@@ -70,26 +91,66 @@ def drawLabel(self):
+            if self.useList:
+                return "matrices = AN.nodes.rotation.c_utils.quaternionListToMatrixList(quaternions)"
+            else:
+                return "matrix = quaternion.normalized().to_matrix().to_4x4()"
         if self.conversionType == "MATRIX_TO_QUATERNION":
-            return "quaternion = matrix.to_quaternion()"
+            if self.useList:
+                return "quaternions = self.matricesToQuaternions(matrices)"
+            else:
+                return "quaternion = matrix.to_quaternion()"
 
         if self.conversionType == "EULER_TO_MATRIX":
-            return "matrix = euler.to_matrix().to_4x4()"
+            if self.useList:
+                return "matrices = AN.nodes.matrix.c_utils.createRotationsFromEulers(eulers)"
+            else:
+                return "matrix = euler.to_matrix().to_4x4()"
         if self.conversionType == "MATRIX_TO_EULER":
-            return "euler = matrix.to_euler('XYZ')"
+            if self.useList:
+                return "eulers = self.matricesToEulers(matrices)"
+            else:
+                return "euler = matrix.to_euler('XYZ')"
 
         if self.conversionType == "QUATERNION_TO_AXIS_ANGLE":
-            if self.useDegree: return "axis, angle = quaternion.axis, math.degrees(quaternion.angle)"
-            else: return "axis, angle = quaternion.to_axis_angle()"
+            if self.useDegree:
+                if self.useList:
+                    return "axises, angles = AN.nodes.rotation.c_utils.quaternionListToAxisListAngleList(quaternions, True)"
+                else:
+                    return "axis, angle = quaternion.axis, math.degrees(quaternion.angle)"
+            else:
+                if self.useList:
+                    return "axises, angles = AN.nodes.rotation.c_utils.quaternionListToAxisListAngleList(quaternions, False)"
+                else:
+                    return "axis, angle = quaternion.to_axis_angle()"
         if self.conversionType == "AXIS_ANGLE_TO_QUATERNION":
-            if self.useDegree: return "quaternion = Quaternion(axis, math.radians(angle))"
-            else: return "quaternion = Quaternion(axis, angle)"
+            if self.useDegree:
+                if self.useList:
+                    return "quaternions = AN.nodes.rotation.c_utils.axisListAngleListToQuaternionList(axises, angles, True)"
+                else:
+                    return "quaternion = Quaternion(axis, math.radians(angle))"
+            else:
+                if self.useList:
+                    return "quaternions = AN.nodes.rotation.c_utils.axisListAngleListToQuaternionList(axises, angles, False)"
+                else:
+                    return "quaternion = Quaternion(axis, angle)"
 
     def getUsedModules(self):
         return ["math"]
+
+    def matricesToQuaternions(self, matrices):
+        return Matrix4x4List.toQuaternions(matrices)
+
+    def matricesToEulers(self, matrices):
+        return Matrix4x4List.toEulers(matrices)
```

Comment:
No need to define those. Embed the calls directly to the execution code.

Reply:
i don't know why my CHANG.MD file alway has a conflicts?

Reply:
so i keep the original file, it is no conflicts now